### PR TITLE
#227 connect hazus curves

### PR DESCRIPTION
**Pull request by Santonia27**
- Updating the linking table with new classification
- update exposure dataframe with new classification
- implement workflow to connect correct damage curve to new classification

PR DelftDashboard: https://github.com/Deltares-research/DelftDashboard/pull/112

### CODE REVIEW

File: hydromt_fiat/workflows/exposure_vector.py [hunk 1]
```diff
@@ -406,38 +406,54 @@ def setup_occupancy_type(
...
+                        self.exposure_db, "Secondary Object Type", "pot"
+                    )
+                elif "Secondary Object Type" in gdf.columns:
+                    gdf.rename(columns={"Secondary Object Type": "pot"}, inplace=True)
+                    self.exposure_db = pd.merge(
+                        self.exposure_db, gdf, on="Object ID", how="left"
+                    )
+                    self.exposure_db = self._set_values_from_other_column(
+                        self.exposure_db, "Secondary Object Type", "pot"
+                    )
```
Comment (LuukBlom) [inline hydromt_fiat/workflows/exposure_vector.py:456]: To make it more intuitive to read and understand, it is good practice to handle exceptions/edge cases first, and then after handling all exceptions, handle the function's responsibility.


```
....
# Update the exposure database
if type_add not in self.exposure_db:
   self.logger.warning(
    "NotImplemented the spatial join of the exposure data with the "
    "occupancy map the for multiple exposure geoms"
    )
    NotImplemented

if ("Primary Object type" not in gdf.columns) and ("Secondary Object Type" not in gdf.columns):
       self.exposure_db = gdf.copy()
       return

if "Primary Object Type" in gdf.columns:
    gdf.rename(columns={"Primary Object Type": "pot"}, inplace=True)
    self.exposure_db = pd.merge(
        self.exposure_db, gdf, on="Object ID", how="left"
    )
    self.exposure_db = self._set_values_from_other_column(
        self.exposure_db, "Primary Object Type", "pot"
    )
    
    # Also rpelace Secondary Object with new classification to assign correct damage curves
    self.exposure_db = pd.merge(
        self.exposure_db, gdf, on="Object ID", how="left"
    )
    self.exposure_db = self._set_values_from_other_column(
        self.exposure_db, "Secondary Object Type", "pot"
    )
elif "Secondary Object Type" in gdf.columns:
    gdf.rename(columns={"Secondary Object Type": "pot"}, inplace=True)
    self.exposure_db = pd.merge(
        self.exposure_db, gdf, on="Object ID", how="left"
    )
    self.exposure_db = self._set_values_from_other_column(
        self.exposure_db, "Secondary Object Type", "pot"
    )
```

Comment (LuukBlom) [reply hydromt_fiat/workflows/exposure_vector.py:456]: Additionally, only one of these blocks will be executed in a function call. 
`elif <expr>` will only be evaluated if all previous if & elif statements in the block were false.
Use 2 `if` statements if you want to do the check for both

```
if "Primary Object Type" in gdf.columns:
    ....
elif "Secondary Object Type" in gdf.columns:
    .....
```

File: hydromt_fiat/api/exposure_vm.py
Comment (LuukBlom) [inline hydromt_fiat/api/exposure_vm.py:1]: Im not sure what the practice is for hydromt_fiat, but function docstrings are quite nice to have in general, especially in the API.
If you have github copilot, you can:
select a function
right click -> Copilot -> generate docs (tell it to use the numpy format), then you just have to read and fix mistakes in there.

**Review by LuukBlom — CHANGES_REQUESTED** [review]

File: hydromt_fiat/workflows/exposure_vector.py (hunk 1 above)
Comment (Santonia27) [reply hydromt_fiat/workflows/exposure_vector.py:456]: I think the logger warning refers to the if-statement some more lines above:
if len(self.exposure_geoms) == 1. But following your example i could update it like this;

`            if type_add not in self.exposure_db:
                    self.exposure_db = gdf.copy()
             else
                if "Primary Object Type" in gdf.columns:
                    gdf.rename(columns={"Primary Object Type": "pot"}, inplace=True)
                    self.exposure_db = pd.merge(
                        self.exposure_db, gdf, on="Object ID", how="left"
                    )
                    self.exposure_db = self._set_values_from_other_column(
                        self.exposure_db, "Primary Object Type", "pot"
                    )
                    
                    # Also rpelace Secondary Object with new classification to assign correct damage curves
                    self.exposure_db = pd.merge(
                        self.exposure_db, gdf, on="Object ID", how="left"
                    )
                    self.exposure_db = self._set_values_from_other_column(
                        self.exposure_db, "Secondary Object Type", "pot"
                    )
                elif "Secondary Object Type" in gdf.columns:
                    gdf.rename(columns={"Secondary Object Type": "pot"}, inplace=True)
                    self.exposure_db = pd.merge(
                        self.exposure_db, gdf, on="Object ID", how="left"
                    )
                    self.exposure_db = self._set_values_from_other_column(
                        self.exposure_db, "Secondary Object Type", "pot"
                    )
            `

Comment (Santonia27) [reply hydromt_fiat/workflows/exposure_vector.py:456]: > Additionally, only one of these blocks will be executed in a function call. `elif <expr>` will only be evaluated if all previous if & elif statements in the block were false. Use 2 `if` statements if you want to do the check for both
> 
> ```
> if "Primary Object Type" in gdf.columns:
>     ....
> elif "Secondary Object Type" in gdf.columns:
>     .....
> ```

 In this case I want to only run one block and not no other if a condition was True beforehand

File: hydromt_fiat/api/exposure_vm.py
Comment (Santonia27) [reply hydromt_fiat/api/exposure_vm.py:1]: Nice call! There's no docstrings in there right now but I will start implementing those!

File: hydromt_fiat/workflows/exposure_vector.py (hunk 1 above)
Comment (LuukBlom) [reply hydromt_fiat/workflows/exposure_vector.py:456]: > I think the logger warning refers to the if-statement some more lines above: if len(self.exposure_geoms) == 1. But following your example i could update it like this;
> 
> ```
>        `if type_add not in self.exposure_db:
>                 self.exposure_db = gdf.copy()
>         else:
>              ...
Oh oops, you are right

**Review by panosatha — APPROVED** [review]
Looks good to me!

**Review by LuukBlom — APPROVED** [review]

**Pull request merged**